### DISCUSSION TITLE
🐯😻 TIGER - Transaction Improvement for Giphy & Emoji Returns - protocol definition draft

### DISCUSSION THREAD

**Discussion by fmhc** [dogecoin/dogecoin#2276] [General]
Hello Community,
in my research about Dogecoin and the Bitcoin technology / protocol and how to enable some kind of smart contracts / vectors to appending something within the existing protocol, i stumbled over the OP_RETURN message which can hold up to 80 Byte of additional data to the transaction information.
This field could be used to store either an additional Hash to identify something on a sidechain, or to hold a key for something which gets activated when a specific wallet is called (you need a running dogecoind and a go script for reading the blocks / tx's in raw and then decode it, this standardized and probably audited (hello, here it gets probably tricky)). Or it could just hold a message, there are some examples (a bitcoin-onchain-forum and a document hash timestamping service, and CryptoGraffiti which is reading links from bitcoin) how this is used already. There is also an interesting discussion in the bitcointalk forum about that - https://bitcointalk.org/index.php?topic=524877.20

I am currently building a test to send a cat over the dogechain and then have a tx browser which is capable of transforming the data back into an image. Thats basic stuff, yes, but how fun would it be to send an https://giphy.com/ gif within an tx. Something like that could be safe to implement (other than allowing random filesize / fileformat images which in the end contain scripts who rob your wallet) and would not involve changes to dogecoin-core; the dogecoin wallet could be forked / upgraded to have a safe way to show the gifs / open a giphy browser tab or there could be a webwallet fork with that addon.

Have there been already discussions about increasing the MaxDataCarrierSize of OP_RETURN or adding an optional, bigger data storage area, to fit adation / integration needs for 2nd / 3rd generation blockchain ecosystem? Some kind of more structured / additional data fields, maybe even encrypted with a second key pair? With the existing protocol, i see only limited smart contract capabilities which would need to be more like plugins which exploit that OP_RETURN message somehow; when not using another chain / a loophole for the data processing - do i miss something? What are proposed architectural thoughts onto that?

I hope for a lot of interesting discussions around this topic, if i missed some project or discussion please link it here, thanks :)

**Comment by fmhc** [comment]
Hello,
we have discussed a more detailled concept here: https://github.com/godcloutier/officialdogevm/discussions/11#discussioncomment-823890
![image](https://user-images.githubusercontent.com/3387385/121914861-f2c52f80-cd32-11eb-8201-5b0a59ac27c3.png)

Please add your thoughts to that

**Comment by fmhc** [comment]
We've experimented a bit further and have successfully send some cat emojis and others on the testnet and already on mainnet - so the Proof of Concept to use the field somehow is done. We're now trying to get it in dogecoin-qt so people can see it rather than dig into raw data, and bring up some other tools to have a basic smart contract wallet which is working somehow based of emojis coming in.

What do we need to add an official dogecoin repository where we define a dogecoin sub-protocol for using this 80 Byte efficiently?
A first draft to start a discussion is here
![image](https://user-images.githubusercontent.com/3387385/122851152-c438ed00-d30e-11eb-97cb-792648850be9.png)
What do you think?

**Comment by michilumin** [comment]
looking at this, @fmhc  - very interesting. Love the idea of using emojis for this so far and the prospect of expanding to memes and gifs. Will look into this more shortly, but ❤️ from me so far. This kind of thing, with a 'dogecoin fun riff' is exactly what I think the community would love to see more of.  Will dive deeper later.

**Reply by fmhc** [reply]
this could be a solution with Qt:
https://forum.qt.io/topic/77986/emoji-support/18
https://github.com/raven-worx/qrwemoticons

**Comment by fmhc** [comment]
🐯 I have started a repository to share some stuff. https://github.com/katzenmalen/dogecoin-tiger - can we discuss that here and then integrate that as a part of the dogecoin project, but seperated from dogecoin-core as dogecoin-tiger? I am looking for anything which can push this thing further.

**Comment by NotIshanSingh** [comment]
Hey could you please update the discord on your git? I have been trying to find a way to contact you as this is an amazing idea!

**Reply by fmhc** [reply]
here is the link: https://discord.gg/zdUFFdQ2Hq - I've updated it on the repo also
